gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/SearchController.groovy

```diff
@@ -26,19 +26,26 @@ import org.springframework.web.bind.annotation.RequestMethod
 import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.RestController
 
-@CompileStatic
+import javax.servlet.http.HttpServletRequest
+
 @RestController
 class SearchController {
   @Autowired
   SearchService searchService
 
   @RequestMapping(value = "/search", method = RequestMethod.GET)
-  List<Map> search(@RequestParam(value = "q") String query,
+  List<Map> search(@RequestParam(value = "q", defaultValue = "", required = false) String query,
```

Comment:
There's a potential performance hit here.  While the UI will set a `pageSize` (to 500, I think).  API users could potentially get a lot more data.  The downside is that we would have to require either a `keyword` or a `name` filter option...